morango/utils/backends/postgres.py

```diff
@@ -2,10 +2,46 @@
 from morango.models import (Buffer, RecordMaxCounter, RecordMaxCounterBuffer,
                             Store)
 
+from django.db import connection
 
 class SQLWrapper(BaseSQLWrapper):
     backend = 'postgresql'
 
+    def _bulk_insert_into_app_models(self, cursor, app_model, fields, db_values, placeholder_str):
+        # cast the values in the SET statement to their appropiate postgres db types
+        set_casted_values = ', '.join(map(lambda f: '{f} = nv.{f}::{type}'.format(f=f.attname, type=f.rel_db_type(connection)), fields))
+        # cast the values in the SELECT statement to their appropiate posgtres db types
+        select_casted_values = ', '.join(map(lambda f: '{f}::{type}'.format(f=f.attname, type=f.rel_db_type(connection)), fields))
+        # cast the pk to a UUID type from a Char type so postgres doesn't complain
+        pk = [f for f in fields if f.primary_key][0]
```

Comment:
What if there are other UUID fields? And other special types? Do we not more generally need to use something like `Field.prepare_value` or whatever it is to do proper conversion? Feels like this could be brittle.

Reply:
sorry the comment may have been a bit misleading since it specifically solves the problem for kolibri, but below we find the proper type
https://github.com/learningequality/morango/pull/66/files/8c6d941fd1e3d9a3ae6a3a894183ebb974994010#diff-9d55d7ca123fd15267bc3f96034b1380R41

Reply:
Yeah, definitely worth updating the comment, as it makes it sound like you are casting all PKs to UUIDs.